packages/transaction-controller/src/TransactionController.ts

```diff
@@ -460,9 +528,12 @@ export class TransactionController extends BaseControllerV1<
       transactions: [],
       lastFetchedBlockNumbers: {},
     };
-
     this.initialize();
-
+    this.hub = hub ?? this.hub;
+    this.enableMultichain = enableMultichain;
+    this.findNetworkClientIdByChainId = findNetworkClientIdByChainId;
```

Comment:
Rather than additional callbacks, would the current best practice be to `call` via the `messenger` object directly to the `NetworkController`?

Reply:
I appreciate the messaging system is more embedded in `BaseControllerV2` but we already have a messaging property available plus the required `TransactionControllerMessenger` type that we're using to communicate with the `ApprovalController`.

Given that, should we not encourage use of the messenger wherever feasible?

Would this not also simplify adoption by reducing the number of needed callbacks?

Reply:
Working on this

Reply:
https://github.com/MetaMask/core/pull/3887